src/Cron_Event_Command.php

```diff
@@ -361,33 +307,50 @@ protected static function run_event( stdClass $event ) {
 	 *     Success: Deleted 2 instances of the cron event 'cron_test'.
 	 */
 	public function delete( $args, $assoc_args ) {
-
-		$hook   = $args[0];
-		$events = self::get_cron_events();
+		$events = self::get_selected_cron_events( $args, $assoc_args );
 
 		if ( is_wp_error( $events ) ) {
 			WP_CLI::error( $events );
 		}
 
 		$deleted = 0;
 		foreach ( $events as $event ) {
-			if ( $event->hook === $hook ) {
-				$result = self::delete_event( $event );
-				if ( $result ) {
-					$deleted++;
-				} else {
-					WP_CLI::warning( sprintf( "Failed to the delete the cron event '%s'.", $hook ) );
-				}
+			$result = self::delete_event( $event );
+			if ( $result ) {
+				$deleted++;
+				WP_CLI::log( sprintf( "Deleted the cron event '%s'", $event->hook ) );
 			}
 		}
 
-		if ( $deleted ) {
-			$message = ( 1 === $deleted ) ? "Deleted the cron event '%2\$s'." : "Deleted %1\$d instances of the cron event '%2\$s'.";
-			WP_CLI::success( sprintf( $message, $deleted, $hook ) );
-		} else {
-			WP_CLI::error( sprintf( "Invalid cron event '%s'.", $hook ) );
+		$message = ( 1 === $deleted ) ? 'Deleted a total of %d cron event.' : 'Deleted a total of %d cron events.';
+		WP_CLI::success( sprintf( $message, $deleted ) );
+	}
+
+	/**
+	 * Executes an event immediately.
+	 *
+	 * @param stdClass $event The event
+	 * @return bool Whether the event was successfully executed or not.
+	 */
+	protected static function run_event( stdClass $event ) {
+
+		if ( ! defined( 'DOING_CRON' ) ) {
+			// phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedConstantFound -- Using native WordPress constant.
+			define( 'DOING_CRON', true );
 		}
 
+		if ( false !== $event->schedule ) {
+			$new_args = array( $event->time, $event->schedule, $event->hook, $event->args );
+			call_user_func_array( 'wp_reschedule_event', $new_args );
```

Comment:
Not sure what's going on here. Why not just this:
```suggestion
			wp_reschedule_event( $event->time, $event->schedule, $event->hook, $event->args );
```

Reply:
Git shows this as change, but this is actually the already existing implementation for the `run_event` command: https://github.com/wp-cli/cron-command/blob/9688e89cf79a75a2ae13acb13af65f81ecdc721c/src/Cron_Event_Command.php#L342

I do agree that your solution makes more sense, and is consistent with how it is done in other places in this class. I'll update it.